### DISCUSSION TITLE
Feature request: Split text input to several outputs by special character

### DISCUSSION THREAD

**Discussion by greek12man** [erew123/alltalk_tts#140] [General]
I would like to request this feature:
On AllTalk TTS Generator

Insert text:
phraseA [split] phraseB [split] phraseC

click generate tts
get 3 wav files as output, each having one of the 3 phrases.

This is a feature that would save me potentially hundreds of hours, instead of generating one by one.
Thank you for your phenomenal work on alltalk and thank you for considering my request.

**Comment by erew123** [comment]
Hi @greek12man 

Are you talking about splitting individual sentences? as you can set the chunk size down to 1 and it will split on each period "." e.g.

`This is three. Sentences that would be generated. As 3 separate generations.`

The above would give you 3x individual wav files:

1) `This is three.` 
2) `Sentences that would be generated.` 
3) `As 3 separate generations.`

Or are you talking about longer multiple sentences that may have periods already in them?

Thanks

**Reply by greek12man** [reply]
Thank you, I have figured this out. Unfortunately sometimes I need to cut not on the period and include two sentences.

**Reply by erew123** [reply]
Humm. I might be able to give you a quick workaround to try. It may or may not break things up as you want, but there's no harm in trying if it helps. So the idea would be to use a semi-colon or a colon in the sentence (I cant exactly recall what all the text splitting is doing right now, but I think this might work. e.g lets say you have:

`This is my first part of my phrase. This is the second part of my phrase. This is a completely different phrase`

So the above you want to generate as 2x phrases:

1) `This is my first part of my phrase. This is the second part of my phrase`
2) `This is a completely different phrase`

So the theory goes that if you introduce a semi-colon, possibly a colon OR indeed 2x semi-colons, it may well give you the break that sounds like it hit a full stop/period. e.g. 

One semi-colon
`This is my first part of my phrase; This is the second part of my phrase. This is a completely different phrase`

Two semi-colon's
`This is my first part of my phrase;; This is the second part of my phrase. This is a completely different phrase`

A colon
`This is my first part of my phrase: This is the second part of my phrase. This is a completely different phrase`

With a chunk size of 1, those should split into the 2x sentences as you want and I **think**, at least hypothetically, may give you the result you want.

Worth a try if nothing else.

**Reply by greek12man** [reply]
This is a very helpful workaround. While testing now I notice it also splits on !
"Yes! We can do this right now; Let's go." was split into two:
Yes!
We can do this right now; Let's go.

I actually have lists with hundreds of entries, some of them have 2/3 sentences. I will have to manually edit the periods to semicolons and redo the ones with exclamation points.
Even if it is not perfect, it still will save me a lot of time.
